src/pages/Swap/LimitOrderBox/components/LimitOrderForm/utils.ts

```diff
@@ -0,0 +1,73 @@
+import { Web3Provider } from '@ethersproject/providers'
+import { formatUnits } from '@ethersproject/units'
+import { CurrencyAmount, TokenAmount } from '@swapr/sdk'
+
+import { SetStateAction } from 'react'
+
+import { getQuote } from '../../api/cow'
+import { SerializableLimitOrder } from '../../interfaces'
+
+export const checkMaxOrderAmount = async (
+  limitOrder: SerializableLimitOrder,
+  setIsPossibleToOrder: (value: SetStateAction<{ status: boolean; value: number }>) => void,
+  setLimitOrder: (value: SetStateAction<SerializableLimitOrder>) => void,
+  amountWei: string,
+  expiresAt: number,
+  sellTokenAmount: TokenAmount,
+  sellCurrencyMaxAmount: CurrencyAmount | undefined,
+  chainId: number,
+  provider: Web3Provider
+) => {
+  const signer = provider.getSigner()
+
+  if (limitOrder.sellToken === limitOrder.buyToken) {
+    return
+  }
+
+  const { quote } = await getQuote({
+    chainId,
+    signer,
+    order: { ...limitOrder, sellAmount: amountWei, expiresAt: expiresAt },
+  })
+
+  if (!quote || !sellCurrencyMaxAmount?.raw) {
```

Comment:
I have couple of comments regarding component and folder structure here.
1) Most of the components and folders inside the swap shouldn't be there. As you can see from other pages  components here should be minimal and contain mostly logic
2) I would say most of the stuff can be moved around or just put into components folder under LimitOrder way better solution then the current one we shouldn't be clogging swapbox folder like this
3) I know its alot of work but just wanted to mention we are not at all following the design patters we outlined like separating UI from logic thats why this structure is messy..
4) There are 2 utils folders inside the LimitOrderBox...

Reply:
I think we need to talk more about this. But I don't see nothing wrong in having components inside the page folder. I think it's better to create and manage components / functions. Everything that is used in multiple pages should be in one generic folder, otherwise we can have it next to where we use it. But we can talk more about this and get our folder structure right.